src/test/java/com/google/sps/servlets/getArticlesServletTest.java

```diff
@@ -0,0 +1,182 @@
+    HttpServletResponse response = mock(HttpServletResponse.class);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    List<Article> articles = new ArrayList<>();
+
+    // Make sure that the size of articles is initially zero before calling the servlet.
+    assertEquals(0,articles.size());
+
+    //calling the servlet
+    new getArticlesServlet().doPut(request, response);
+
+    //verify that request was never accessed
+    verify(request, never()).getParameter(anyString());
+
+    Query query = new Query("Article");
+    PreparedQuery results = datastore.prepare(query);
+    
+    for (Entity entity : results.asIterable()) {
+      String entityKey = KeyFactory.createKeyString(entity.getKey().getKind(), entity.getKey().getId());
+      String publisher = (String) entity.getProperty("publisher");
+      String author = (String) entity.getProperty("author");
+      String title = (String) entity.getProperty("title");
+      String description = (String) entity.getProperty("description");
+      String url = (String) entity.getProperty("url");
+      String publishedAt = (String) entity.getProperty("publishedAt");
+
+      Article newArticle = new Article(entityKey, publisher, author, title, description, url, publishedAt);
+      articles.add(newArticle);
+    }
+
+    // After calling the API, the size of articles can never be zero. 
+    assertNotEquals(0,articles.size());
+
+  }
+
+  @Test
+  public void deleteResultsOfQueryFromDatastoreTest() throws Exception {
+
+    //mocking response and request to pass to servlet.
+    HttpServletRequest request = mock(HttpServletRequest.class);
+    HttpServletResponse response = mock(HttpServletResponse.class);
+
+    DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+
+    Entity articleEntity = new Entity("Article");
+    articleEntity.setProperty("publisher", "testPublisher");
+    articleEntity.setProperty("author", "testAuthor");
+    articleEntity.setProperty("title", "testTitle");
+    articleEntity.setProperty("description", "testDescription");
+    articleEntity.setProperty("url", "testUrl");
+    articleEntity.setProperty("publishedAt", "testPublishedAt");
+
+    Entity articleEntity1 = new Entity("Article");
+    articleEntity1.setProperty("publisher", "testPublisher2");
+    articleEntity1.setProperty("author", "testAuthor2");
+    articleEntity1.setProperty("title", "testTitle2");
+    articleEntity1.setProperty("description", "testDescription2");
+    articleEntity1.setProperty("url", "testUrl2");
+    articleEntity1.setProperty("publishedAt", "testPublishedAt2");
+    
+    // Adding two entities to our mock datastore
+    ds.put(articleEntity);
+    ds.put(articleEntity1);
+
+    Query query = new Query("Article");
+    PreparedQuery results = ds.prepare(query);
+
+    // Checking if the datastore has these two entities. 
+    assertEquals(2,results.countEntities());
```

Comment:
I think this test is unnecessary. Going from the Datastore API you know that when you put two entities of a specified type into datastore, you will retrieve them by querying for that specified entity. I just feel like this check is testing if datastore is working correctly.

Reply:
This is just to make sure the objects are correctly put into the mock datastore. I wanted to check this before checking if it's zero after we call the delete function, since I wanted to know if the two article entities are correctly put into the datastore.

Reply:
Ok, not a problem.